src/client/gui/lib/close_terminal_dialog.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart' hide Switch;
+
+import 'confirmation_dialog.dart';
+import 'switch.dart';
+
+class CloseTerminalDialog extends StatefulWidget {
+  final Function() onYes;
+  final Function() onNo;
+  final Function(bool doNotAsk) onDoNotAsk;
+
+  const CloseTerminalDialog({
+    super.key,
+    required this.onYes,
+    required this.onNo,
+    required this.onDoNotAsk,
+  });
+
+  @override
+  State<CloseTerminalDialog> createState() => _CloseTerminalDialogState();
+}
+
+class _CloseTerminalDialogState extends State<CloseTerminalDialog> {
+  var doNotAsk = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return ConfirmationDialog(
+      title: 'Are you sure you want to close this terminal?',
+      body: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
+        const Text('Its current state will be lost.'),
+        const SizedBox(height: 12),
+        Switch(
+          value: doNotAsk,
+          label: 'Do not ask me again.',
```

Comment:
For another time: I would love to be able to click on the label/area before a switch to have it toggled (not have to place the mouse exactly on top of th widget). This would apply to other switches, check boxes... radio buttons when enabling... WDYT? Does Flutter have a mechanism to achieve that easily?

Reply:
Yes, that can be easily achieved. And in places like this popup, it would indeed make a lot of sense.
But I'm not sure about the settings page, because there you also have dropdowns and text fields, for which I don't know if this pattern makes sense, and having different actionable areas based on the type of setting would also not be nice. Of course, we can also make this aspect configurable, and have places where the whole label is clickable and places where its not, but then that would be introducing an inconsistency for the sake of removing another, so I don't know...

Reply:
Hmm I see what you mean. I think I am used to being able to click on checkbox labels, not sure about the others. Maybe in the settings we'd just need a bigger switch button. Anyway, just something to keep in the back of our heads for later I guess.